### ISSUE TITLE
Support Windows

### ISSUE THREAD

**Issue by ewjoachim** [procrastinate-org/procrastinate#491]
- [ ] Add windows tests in the CI
- [ ] Check that signals work the same way
- [ ] Check that envvars work the same way
- [ ] Adjust documentation
- [ ] Find a windows maintainer

**Comment by ewjoachim** [comment]
Ping @aleksandr-shtaub

**Comment by aleksandr-shtaub** [comment]
I'll start from the easy ones.

### Env vars

Works as expected.

#### CMD

```cmd
rem Set value for a variable Name. Note: Should not contain any whitespaces around `=`.
> set Name=Value

rem Get value of variable `Name`.
> echo %Name%
Value
```

or 

#### PowerShell

```powershell
# Set value for a variable Name. Note: powershell have types, so use `""` for a string type.
> $Env:Name = "Value"  

# Get value of variable.
> echo $Env:Name
Value
```

#### Example

```python
# test.py
import os
print(os.environ.get("Greeter"))
```

```powershell
> $Env:Greeter = "Hello!"
> python -m test
Hello!
```

### Paths

Unix: `a/b/c`
Win: `a\\b\\c`

Consider using `Path.asPosix()` instead of `str(Path)` when testing paths.

### Signals

`signal.SIGINT`, `signal.SIGTERM` are supported. 
Refer: <https://docs.python.org/3/library/signal.html#signal.signal>

### Even Loop

- `add_signal_handler()` and `remove_signal_handler()` aren't supported on win. Refer: <https://docs.python.org/3/library/asyncio-eventloop.html#unix-signals> 
- We cant' use `ProactorEventLoop` because `add_reader()` and `add_writer()` are used by aiopg.
- Since Python 3.8 we need to change default policy explicitly like:

```python
if os.platform == "win32":
    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())
```

Refer: <https://docs.python.org/3/library/asyncio-platforms.html>

TBC (if I find something else)

**Comment by aleksandr-shtaub** [comment]
Related:
#286
#444 
#451

**Comment by ewjoachim** [comment]
Or even https://github.com/procrastinate-org/procrastinate/labels/Issue%20contains%3A%20Some%20Windows :)

**Comment by spapas** [comment]
Hello friends, can you please clarify if it is possible to run this project under windows? This isn't clear for me right now. I only care about a test/dev environment (my production is on linux). If it is possible can you recommend me how ?

Right now, when trying to run the worker under windows I'm getting the following error:

```
python manage.py procrastinate worker
error connecting in 'pool-1': Psycopg cannot use the 'ProactorEventLoop' to run in async mode. Please use a compatible event loop, for instance by setting 'asyncio.set_event_loop_policy(WindowsSelectorEventLoopPolicy())'
```

**Comment by ewjoachim** [comment]
It's untested. If it works, it's mainly by chance, and if it doesn't but you see how to fix it, we accept pull requests. If anyone is willing to step up and maintain the lib under Windows (e.g. accept to be pinged whenever something comes up), I'll be happy to share the keys and call the project officially supporting Windows :)

I don't have a windows machine (well, not really).

**Comment by spapas** [comment]
I did some quick tests and found out that two changs are needed for this to run on windows (at least for dev):

1. Add this line `asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())` before running the procrastinate cli
2. Do not use signals in signals.py

1 is simple however I can't understand the implications of 2. Let's suppose we don't do anything on `install` and `uninstall` functions of signals.py what would happen? Can you ELI5 what's the purpose of these signal handlers?

TIA

**Comment by ewjoachim** [comment]
Signals are used so that when the (Unix-based) OS informs procrastinate it needs to stop gracefully. Without signal, when you interrupt (ctrl+c, sigint) the program, or the process manager wants to stop it (sigterm), it would stop immediately, halfway through running jobs. Thanks to signals instead, we continue running but stop taking new jobs and halts when all running jobs are done (modulo various timeouts)

**Comment by spapas** [comment]
@ewjoachim thank youfor the explanation. As I understand this project can be used for *development* in Windows with very minor changes (apply the `WindowsSelectorEventLoopPolicy` and ignore signals). Would it be possible to implement that functionality in a new release?

**Comment by ewjoachim** [comment]
> Would it be possible to implement that functionality in a new release?

It would be possible, but this doesn't say who does it :) 

I'm running my own contribution time on a limited schedule at the moment, so it's somehow complicated for me to commit on doing the PR, especially since I wouldn't be able to test that it works correctly. On the other hand, of you want to submit a PR, it will be much easier for me (or another maintainer) to review and merge it 😃

**Comment by spapas** [comment]
Hello @ewjoachim I understand the problem. If you are willing to accept such a PR, i.e no proper windows support but support for development under windows I'd be happy to test the needed changes on my development environment and provide a PR.
